backend/siarnaq/api/compete/serializers.py

```diff
@@ -61,10 +63,36 @@ def update(self, instance, validated_data):
         return instance
 
 
+@deconstructible
+class FileValidator:
+    error_messages = {
+        "max_size": (
+            "Ensure this file size is not greater than %(max_size)s."
+            " Your file size is %(size)s."
+        ),
+    }
+
+    def __init__(self, max_size=None):
+        self.max_size = max_size
+
+    def __call__(self, data):
+
+        if self.max_size is not None and data.size > self.max_size:
+            params = {
+                "max_size": filesizeformat(self.max_size),
+                "size": filesizeformat(data.size),
+            }
+            raise serializers.ValidationError(self.error_messages["max_size"], params)
+
+    def __eq__(self, other):
+        return isinstance(other, FileValidator) and self.max_size == other.max_size
+
+
 class SubmissionSerializer(serializers.ModelSerializer):
     teamname = serializers.CharField(source="team.name", read_only=True)
     username = serializers.CharField(source="user.username", read_only=True)
-    source_code = serializers.FileField(write_only=True)
+    validate_file = FileValidator(max_size=5 * 1024 * 1024)
```

Comment:
I think it's confusing to have this as a class variable if it's only needed for defining the line below. Happy if you just inline this into the below line, or any other alternate solution where we don't have an extra class variable

Reply:
Agree with this- otherwise all looks good to me!